### ISSUE TITLE
Judgment exception for two completely equivalent loop operations.

### ISSUE THREAD

**Issue by Elowen-jjw** [seahorn/seahorn#493]
```
#include "seahorn/seahorn.h"

int main(){
    int x = 0;
    int y = x;
    for(int i=0; i<10; i++){
        x += i;
    }
    for(int i=0; i<10; i++){
        y += i;
    }
    sassert(x==y);
}

```

In this example, why seahorn gives the "sat" result? Thanks.

Version: seahorn 14.0.0
OS: ubuntu 22.04
Command line:
```
sea pf -m64 example.c
```

**Comment by caballa** [comment]
Same problem described [here](https://github.com/seahorn/seahorn/issues/492)

**Comment by agurfinkel** [comment]
There was a bug that did not disable vectorization by default. This is fixed in commit 8bfaa645

Note that SeaHorn is not great at this kind of a problem. It will not learn a quantified invariant by default. So it will have to enumerate every array entry. This, of course, requires unrolling the array.

Moreover, SeaHorn is pretty bad for arrays with fixed bound. These are unrolled (either statically or dynamically). It is better to abstract exact number of iterations by a symbolic constant.
